lib/api/v3/memberships/memberships_api.rb

```diff
@@ -35,7 +35,9 @@ class MembershipsAPI < ::API::OpenProjectAPI
         resources :memberships do
           get &::API::V3::Utilities::Endpoints::Index
                  .new(model: Member,
-                      scope: -> { Member.includes(MembershipRepresenter.to_eager_load) },
+                      # TODO: For now we exclude entity specific memberships in the API until we have updated the
+                      # frontend and representers to show those properly
+                      scope: -> { Member.where(entity: nil).includes(MembershipRepresenter.to_eager_load) },
```

Comment:
This fixes the issue for the index action. The show endpoint does still return an entity membership if the right id is provided.